src/analyze-trace-dir.ts

```diff
@@ -21,15 +22,16 @@ const argv = yargs(process.argv.slice(2))
         .strict())
     .argv;
 
-const limit = plimit(os.cpus().length);
+// Try to leave one core free
+const limit = plimit(Math.max(1, os.cpus().length - 1));
 
 const traceDir = argv.traceDir!;
 
 main().then(
-    code => process.exit(code),
+    code => process.exitCode = code,
     err => {
         console.error(`Internal error: ${err.message}`);
-        process.exit(2);
+        exit(2);
     });
```

Comment:
I'm curious, why not `exit(code)` in the then to be consistent, and why not `then`/`catch`?

```ts
main()
  .then(code => exit(code))
  .catch(err => {
    console.error(`Internal error: ${err.message}`);
    exit(2);
  });
```

Reply:
If there's a reason for not splitting out the `catch`, I don't remember it.